pymc3/tests/test_distributions.py

```diff
@@ -553,6 +553,30 @@ def RandomPdMatrix(n):
     return np.dot(A, A.T) + n * np.identity(n)
 
 
+def test_hierarchical_logpt():
+    """Make sure there are no random variables in a model's log-likelihood graph."""
+    with pm.Model() as m:
+        x = pm.Uniform("x", lower=0, upper=1)
+        y = pm.Uniform("y", lower=0, upper=x)
+
+    logpt_ancestors = list(ancestors([m.logpt]))
+    ancestors_with_owner = [a for a in logpt_ancestors if a.owner]
+    assert len(ancestors_with_owner) > 0
+    assert not any(isinstance(v.owner.op, RandomVariable) for v in ancestors_with_owner)
+    assert x.tag.value_var in logpt_ancestors
+    assert y.tag.value_var in logpt_ancestors
+
+
+def test_hierarchical_obs_logpt():
+    obs = np.array([0.5, 0.4, 5, 2])
+
+    with pm.Model() as model:
+        x = pm.Normal("x", 0, 1, observed=obs)
+        pm.Normal("y", x, 1, observed=obs)
+
+    model.logp(model.test_point)
```

Comment:
What is the test here, that it computes the test point?

Reply:
This test is unfinished (and it was essentially copied from somewhere else), but it makes sure that the `Observed`-typed `x` can be used properly as the input to another variable's log-likelihood.

The next commit I'm going to push gets rid of `Observed`, though, so the problem that this test highlights won't matter after that.